tests/integration/test_persist_payload.py

```
@@ -0,0 +1,246 @@
+            taskId=self.sample_http_payload_data["taskId"],
+            question=question_model,
+            response=response_model,
+        )
+
+        # Simulate a MongoDB exception during the insert operation
+        with patch.object(
+            self.mock_client["hivemind"]["http_messages"],
+            "insert_one",
+            side_effect=Exception("Database error"),
+        ):
+            with self.assertLogs(level="ERROR") as log:
```

Ruff

Use a single `with` statement with multiple contexts instead of nested `with` statements

Combine `with` statements

(SIM117)